### PR TITLE
ref(js): Convert CreateTeamModal to a FC

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

**Review by scttcper — APPROVED** [review]

File: static/app/components/modals/createTeamModal.tsx [hunk 1]
```diff
@@ -1,52 +1,47 @@
...
+    data: {slug: string},
+    onSuccess: Function,
+    onError: Function
+  ) {
+    try {
+      const team: Team = await createTeam(api, data, {orgId: organization.slug});
+
+      closeModal();
+      onClose?.(team);
+      onSuccess(team);
```
Comment (priscilawebdev) [inline static/app/components/modals/createTeamModal.tsx:31]: is this order fine?

File: static/app/components/modals/createTeamModal.tsx [hunk 2]
```diff
@@ -1,52 +1,47 @@
...
 import {Client} from 'app/api';
 import CreateTeamForm from 'app/components/teams/createTeamForm';
 import {t} from 'app/locale';
 import {Organization, Team} from 'app/types';
-import withApi from 'app/utils/withApi';
+import useApi from 'app/utils/useApi';
 
-type Props = {
+type Props = ModalRenderProps & {
   api: Client;
```
Comment (priscilawebdev) [inline static/app/components/modals/createTeamModal.tsx:12]:
```suggestion
```

File: static/app/components/modals/createTeamModal.tsx [hunk 3]
```diff
@@ -1,52 +1,47 @@
...
-      });
-  };
-
-  handleSuccess(team: Team) {
-    if (this.props.onClose) {
-      this.props.onClose(team);
+};
+
+function CreateTeamModal({Body, Header, ...props}: Props) {
+  const {onClose, closeModal, organization} = props;
```
Comment (matejminar) [inline static/app/components/modals/createTeamModal.tsx:18]: Should these be destructured in one place?

**Review by matejminar — APPROVED** [review]

File: static/app/components/modals/createTeamModal.tsx (hunk 3 above)
Comment (evanpurkhiser) [reply static/app/components/modals/createTeamModal.tsx:18]: We pass down the `props`, which is why I did this.

File: static/app/components/modals/createTeamModal.tsx (hunk 2 above)
Comment (evanpurkhiser) [reply static/app/components/modals/createTeamModal.tsx]: yep missed that, ty

**Pull request merged**